File: dataStruct/Tree.cpp
#include<iostream>
using namespace std;


class Node{
    public:
        int data;
        Node *left;
        Node *right;
};

Node *CreateNode(int data){
    Node *newNode = new Node();
    newNode->left = newNode->right =nullptr;
    return newNode;
}
//Preorder traversing
void printTreePre(Node *root){
    if(root == nullptr)
        return;
    cout<< root->data <<endl;
    printTreePre(root->left);
    printTreePre(root->right);
}
//inorder traversing
void printTreeIn(Node *root){
    if(root == nullptr)
        return;    
    printTreeIn(root->left);
    cout<< root->data <<endl;
    printTreeIn(root->right);

}
//Postorder traversing
void printTreePost(Node *root){
    if(root == nullptr)
        return;    
    printTreePost(root->left);
    printTreePost(root->right);
    cout<< root->data <<endl;
}

int main(){
     //Level 0
     Node *root = CreateNode(1);
     //Level 1
     root->left = CreateNode(2);
     root->right = CreateNode(3);     
     //Level 2
     root->left->left = CreateNode(4);
     root->left->right = CreateNode(5);
     root->right->left = CreateNode(6);
     root->right->right =CreateNode(7);     
     //Level 3
     root->left->right->left = CreateNode(9);
     root->right->right->left = CreateNode(15);
     
     printTreePre(root);
    return 0;
}

File: python/DataStructure/tree.py

class Node(object):
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None
    def insert(self, data):
        if self.data is None:
            self.data = data
        else:
            if data < self.data:
                if self.data is not None:
                    self.left.insert(data)
                else:
                    self.left.Node(data)
            elif data > self.data:
                if self.right is None:
                    self.right = Node(data)
                else:
                    self.right.insert(data)

    def printTree(self):
        if self.left:
            self.left.printTree()
        print(self.data)
        if self.right:
            self.right.printTree()
def preorder(root):
    if root:
        print(root.data)
        preorder(root.left)
        preorder(root.right)
def inorder(root):
    if root:
        inorder(root.left)
        print(root.data)
        inorder(root.right)
def postorder(root):
    if root:
        postorder(root.left)
        postorder(root.right)
        print(root.data)





def dataValue():
    root = Node(50)
    root.insert(20)
    root.insert(11)
    root.insert(15)
    root.insert(45)
    root.insert(30)
    root.insert(78)
    root.printTree()
def main():
    dataValue()

if __name__=='__main__':
    main()

